apps/namadillo/src/App/Transfer/TransferModule.tsx

```diff
@@ -144,27 +146,27 @@ export const TransferModule = ({
 
   const availableAmountMinusFees = useMemo(() => {
     const { selectedAssetAddress, availableAmount } = source;
+
     if (
       typeof selectedAssetAddress === "undefined" ||
-      typeof availableAmount === "undefined"
+      typeof availableAmount === "undefined" ||
+      typeof source.availableAssets === "undefined"
     ) {
       return undefined;
     }
 
     if (
-      !displayGasFee ||
-      !displayGasFee.totalDisplayAmount ||
+      !displayGasFee?.totalDisplayAmount ||
       // Don't subtract if the gas token is different than the selected asset:
-      (gasConfig?.gasToken &&
-        isTransparentAddress(gasConfig.gasToken) &&
-        gasConfig.gasToken !== selectedAssetAddress)
+      gasConfig?.gasToken !== selectedAssetAddress
     ) {
       return availableAmount;
     }
 
-    const amountMinusFees = availableAmount.minus(
-      displayGasFee.totalDisplayAmount
-    );
+    const amountMinusFees = availableAmount
+      .minus(displayGasFee.totalDisplayAmount)
+      .decimalPlaces(6);
```

Comment:
shouldn't we apply this only on render components to be safer?

Reply:
I think we're fine, since we won't have more than 6 decimal places. In the worst case we'll have a difference of less than 0.000001 for a given amount